src/main/java/org/openrewrite/java/testing/mockito/SimplifyMockitoVerifyWhenGiven.java

```diff
@@ -0,0 +1,122 @@
+                     List<Expression> updatedArguments = new ArrayList<>(originalArguments);
+                    updatedArguments.set(0, updatedInvocation);
+
+                    return mi.withArguments(updatedArguments);
+                } else if (isInvokedOnVerify(mi)) {
+                    return checkAndUpdateEq(mi);
+                }
+
+                return mi;
+            }
+
+            private boolean isMockitoWhen(J.MethodInvocation methodInvocation) {
+                return methodInvocation.getMethodType() != null &&
+                       TypeUtils.isOfType(methodInvocation.getMethodType().getDeclaringType(), JavaType.buildType("org.mockito.Mockito")) &&
+                       methodInvocation.getSimpleName().equals("when");
+            }
+
+            private boolean isInvokedOnVerify(J.MethodInvocation methodInvocation) {
+                if (methodInvocation.getSelect() == null) {
+                    return false;
+                }
+                 Expression select = methodInvocation.getSelect();
+
+                if (!(select instanceof J.MethodInvocation) || ((J.MethodInvocation) select).getMethodType() == null) {
+                    return false;
+                }
+
+                 J.MethodInvocation selectInvocation = (J.MethodInvocation) select;
+                return TypeUtils.isOfType(selectInvocation.getMethodType().getDeclaringType(), JavaType.buildType("org.mockito.Mockito")) &&
+                       selectInvocation.getSimpleName().equals("verify");
+            }
+
+            private J.MethodInvocation checkAndUpdateEq(J.MethodInvocation methodInvocation) {
+                 List<Expression> originalArguments = methodInvocation.getArguments();
+                 boolean onlyEqArguments = originalArguments.stream().allMatch(this::isEqExpression);
+                if (!onlyEqArguments) {
+                    return methodInvocation;
+                }
+
+                 List<Expression> updatedArguments = originalArguments.stream()
+                        .map(J.MethodInvocation.class::cast)
+                        .map(invocation -> invocation.getArguments().get(0).<Expression>withPrefix(invocation.getPrefix()))
+                        .collect(Collectors.toList());
+
+                return methodInvocation.withArguments(updatedArguments);
+            }
+
+            private boolean isEqExpression(Expression expression) {
+                if (!(expression instanceof J.MethodInvocation)) {
+                    return false;
+                }
+                 J.MethodInvocation invocation = (J.MethodInvocation) expression;
+
+                if (invocation.getMethodType() == null || !TypeUtils.isOfType(invocation.getMethodType().getDeclaringType(), JavaType.buildType("org.mockito.ArgumentMatchers"))) {
+                    return false;
+                }
+
+                return invocation.getSimpleName().equals("eq");
```

Comment:
Lets [use MethodMatchers](https://docs.openrewrite.org/reference/method-patterns#authoring-recipes-with-methodmatcher) instead of comparing the simple name everywhere we can. :)

Reply:
Ah, that is a lot more ergonomic, thank you!